_posts/builtin/2017-10-23-nvidiatm-triton-server-inference-engine.md

```diff
@@ -0,0 +1,170 @@
+ - **Local Nvidia Triton Server**: false
+
+## AI Model Encryption support
+
+For ensuring inference integrity and providing copyright protection of deep-learning models on edge devices, Kura provides decryption capabilites for trained models to be served through the Triton Server.
+
+### How it works
+
+**Prerequisites**: a deep-learning trained model (or more) exists with the corresponding necessary configuration for running on the Triton Server without encryption. A folder containing the required files (model, configuration etc) has been tested on a Triton Server.
+
+**Restrictions**: if model encryption is used, the following restrictions apply:
+  - model encryption support is only available for a *local* Triton Server instance
+  - all models in the folder containing the encrypted models *must* be encrypted
+  - all models *must* be encrypted with OpenPGP-compliant AES 256 cipher algorithm
+  - all models *must* be encrypted with the same password
+
+Once the development of the deep-learning model is complete, the developer who wants to deploy the model on the edge device in a secure manner can proceed with encrypting the Triton model using the procedure detailed below. After encrypting the model he/she can transfer the file on the edge device using his/her preferred method.
+
+![model_encryption_workflow]({{ site.baseurl }}/assets/images/builtin/model_encryption_workflow.png)
+
+Kura will keep the stored model protected at all times and have the model decrypted **in runtime only** for use by the Inference Server Runtime. As soon as the model is correctly loaded into memory the decrypted model will be removed from the filesystem.
+
+As an additional security measure, the [Model Repository](https://github.com/triton-inference-server/server/blob/main/docs/model_repository.md) containing the decrypted models will be stored in a temporary subfolder and will feature restrictive permission such that only Kura, the Inference Server and the `root` user will be able to access it.
+
+### Encryption procedure
+
+Given a trained model inside the folder `tf_autoencoder_fp32` (for example) with the following layout (see the [official documentation](https://github.com/triton-inference-server/server/blob/main/docs/model_repository.md) for details):
+
+```
+tf_autoencoder_fp32
+├── 1
+│   └── model.savedmodel
+│       ├── assets
+│       ├── keras_metadata.pb
+│       ├── saved_model.pb
+│       └── variables
+│           ├── variables.data-00000-of-00001
+│           └── variables.index
+└── config.pbtxt
+```
+
+Compress the model into a zip archive with the following command:
+
+```bash
+zip -vr tf_autoencoder_fp32.zip tf_autoencoder_fp32/
+```
+
+then encrypt it with the AES 256 algorithm using the following `gpg` command:
+
+```bash
+gpg --armor --symmetric --cipher-algo AES256 tf_autoencoder_fp32.zip
+```
+
+The resulting archive `tf_autoencoder_fp32.zip.asc` can be transferred to the target machine and will be decrypted by Kura.
```

Comment:
Maybe we can mention explicitly that the encrypted model must be placed in the directory configured as **Local model repository path**.